File: algo/problems/0242-valid-anagram/python/valid_anagram.py
from collections import Counter


def is_anagram(s: str, t: str) -> bool:
    """字符计数相等即互为字母异位词。O(n)/O(k)，k=字符集大小。
    Counter 让它变成一行：先比长度（快速剪枝），再比计数。"""
    if len(s) != len(t):
        return False
    return Counter(s) == Counter(t)

File: algo/problems/0242-valid-anagram/js/validAnagram.js
/**
 * 字符计数：用 Map 统计 s，再用 t 抵消。O(n)/O(k)。
 * @param {string} s
 * @param {string} t
 * @returns {boolean}
 */
function isAnagram(s, t) {
  if (s.length !== t.length) return false;
  const count = new Map();
  for (const c of s) count.set(c, (count.get(c) ?? 0) + 1);
  for (const c of t) {
    const n = count.get(c);
    if (!n) return false; // undefined 或 0 都说明 t 多出了字符
    count.set(c, n - 1);
  }
  return true;
}

module.exports = { isAnagram };
